bin/pcl/src/auth.rs

```diff
@@ -0,0 +1,165 @@
+use dirs::home_dir;
+use eyre::Result;
+use reqwest::Client;
+use serde::{Deserialize, Serialize};
+use std::fs;
+use std::path::PathBuf;
+use tokio::time::{sleep, Duration};
+
+const AUTH_FILE: &str = ".phylax/auth.json";
+const BASE_URL: &str = "https://credible-layer-dapp.pages.dev";
+const POLL_INTERVAL: Duration = Duration::from_secs(2);
+const MAX_RETRIES: u32 = 150; // 5 minutes worth of 2-second intervals
+
+const PHYLAX_ASCII: &str = r#"
+@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+@@@@@@@@@@@@@@@@@@BD>"            "<8@@@@@@@@@@@@@@@@@@@
+@@@@@@@@@@@@@@BP"     _.               '4@@@@@@@@@@@@@@@
+@@@@@@@@@@@@D    _e@@B  ,_,   __  t@g_    `G@@@@@@@@@@@@
+@@@@@@@@@@P   _g@@@@P  /@@@  [@@@  \@@@@_    %@@@@@@@@@@
+@@@@@@@@B   _B@@@@@W  {@@@@  [@@@@  T@@@@@a   `@@@@@@@@@
+@@@@@@@P   g@@@@@@@  ;@@@@@  [@@@@A  @@@@@@@_   f@@@@@@@
+@@@@@@P  ,@@@@@@@@F  @@@@@@  g@@@@@  !@@@@@@@L   V@@@@@@
+@@@@@B   @@@@@@@@@  ;@@@@@@  B@@@@@|  @@@@@@@@L   @@@@@@
+@@@@@'  [@@@@@@@@@  g@@BBD>  <4B@@@@  @@@@@@@@@   '@@@@@
+@@@@@   @@@@@@@@@@  BW  __    __ `8@  B@@@@@@@@j   @@@@@
+@@@@@                 ;@@@@  B@@@;                 @@@@@
+@@@@@   qgg@@@@@@g  __ "B@@  @BB" __  g@@@@@@gq;   @@@@@
+@@@@@   @@@@@@@@@@  @@@q___  ___g@@B  @@@@@@@@@   .@@@@@
+@@@@@@   @@@@@@@@@  [@@@@@@  @@@@@@|  @@@@@@@@P   g@@@@@
+@@@@@@\  '@@@@@@@@,  @@@@@@  @@@@@@  |@@@@@@@W   /@@@@@@
+@@@@@@@L  `@@@@@@@@  0@@@@g  @@@@@F  @@@@@@@P   /@@@@@@@
+@@@@@@@@p   \@@@@@@,  @@@@8  @@@@W  A@@@@@B    j@@@@@@@@
+@@@@@@@@@@_   "@@@@@,  @@@]  @@@D  /@@@@D    _@@@@@@@@@@
+@@@@@@@@@@@@_    <B@@_  <=   "8"  /@BP"    _@@@@@@@@@@@@
+@@@@@@@@@@@@@@@_     ""          "      _g@@@@@@@@@@@@@@
+@@@@@@@@@@@@@@@@@@g__              __g@@@@@@@@@@@@@@@@@@
+@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+"#;
+
+#[derive(Serialize, Deserialize)]
+struct AuthResponse {
+    code: String,
+    sessionId: String,
+    deviceSecret: String,
+    expiresAt: String,
+}
+
+#[derive(Serialize, Deserialize)]
+struct StatusResponse {
+    verified: bool,
+    address: Option<String>,
+    token: Option<String>,
+    refresh_token: Option<String>,
+}
+
+#[derive(Serialize, Deserialize)]
+struct StoredAuth {
+    access_token: String,
+    refresh_token: String,
+    address: String,
+}
+
+pub async fn login() -> Result<()> {
+    let client = Client::new();
+
+    // Get authentication code
+    let auth_response: AuthResponse = client
+        .get(format!("{}/api/v1/cli/auth/code", BASE_URL))
+        .send()
+        .await?
+        .json()
+        .await?;
+
+    // Display code and instructions to user
+    println!("\n🔑 Your authentication code is:\n");
```

Comment:
We don't use println multiple times.

It's better to use a big string such as """ sfa""" that allows multi line strings or use \n in the string